### PR TITLE
Feature/ng pagination

### PR DESCRIPTION
**Pull request by superNazo**
[Link to JIRA ticket](https://ssu-jira.softserveinc.com/browse/LR-184)

### Author.
Nazarii Ivankiv

- [x] I have added link to this PR to the JIRA
- [x] I have logged time in JIRA.

### Reviewers.

- [x] @YUzhva 
- [x] @anatoliiibrahimov 
- [x] @vovka

### Issue summary.

Added Pagination to Quizzes list page

### Testing approach.

Manual testing

### CODE REVIEW

File: app/controllers/quizzes_controller.rb [hunk 1]
```diff
@@ -4,7 +4,9 @@ class QuizzesController < ApplicationController
 
   # GET /quizzes
   def index
-    @quizzes = Quiz.all.paginate(page: params[:page], per_page: 10)
+    collection = Quiz.all.paginate(page: params[:page], per_page: 10)
+    @quizzes = QuizzesViewObject.new(collection)
+    # @total_pages = ((@quizzes.count % 10) ? (@quizzes.count / 10) + 1 : (@quizzes.count / 10))
```
Comment (vovka) [inline app/controllers/quizzes_controller.rb]: Codacy Issue found: [Line is too long. [96/80]](https://www.codacy.com/app/vovka/video_quiz/file/2558096576/issues/source?bid=3205144&fileBranchId=3292527#l9)

File: Gemfile [hunk 2]
```diff
@@ -19,14 +19,15 @@ gem 'uglifier',       '2.5.3'
...
+gem 'angular-ui-bootstrap-rails'
 
 group :development, :test do
   gem 'byebug',      '3.4.0'
   gem 'jasmine'
   gem 'rspec-rails', '3.4.0'
   gem 'spring',      '1.1.3'
   gem 'sqlite3',     '1.3.9'
-  gem 'web-console', '2.0.0.beta3'
+  gem 'web-console', '2.0.0'
```
Comment (vovka) [inline Gemfile]: This shouldn't be here.

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 3]
```diff
@@ -28,9 +28,16 @@ quizzesControllers.controller('newQuizCtrl', ['$scope', '$http', '$location', 'Q
 
 quizzesControllers.controller('indexQuizCtrl',
   ['$scope', '$location', 'Quizzes', 'Quiz',
-  function($scope, $location, Quizzes, Quiz ) {
+  function($scope, $location, Quizzes, Quiz) {
+    $scope.total_items = 14;
+    $scope.currentPage = 1;
+    $scope.maxSize = 5;
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: What are these digits^^^^^?

File: app/assets/javascripts/angularjs_app/components/quizzes/views/indexQuiz.html [hunk 4]
```diff
@@ -15,3 +15,6 @@
   </li>
 </ul>
 <a href="/quizzes/new" class="link-to btn btn-success right">New quiz</a>
+
+<uib-pagination ng-change="setPage()" boundary-links="true" total-items="total_items" ng-model="currentPage" class="pagination-sm" previous-text="&lsaquo;" next-text="&rsaquo;" first-text="&laquo;" last-text="&raquo;"></uib-pagination>
+<pre>Page: {{currentPage}} / {{numPages}}</pre>
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/views/indexQuiz.html]: Where did `numPages` come from?

File: app/controllers/quizzes_controller.rb (hunk 1 above)
Comment (vovka) [inline app/controllers/quizzes_controller.rb]: Remove this pls

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -12,6 +12,7 @@
     </script>
     <%= csrf_meta_tags %>
     <base href="/" />
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/angular-ui-bootstrap/1.3.2/ui-bootstrap-tpls.js"></script>
```
Comment (vovka) [inline app/views/layouts/application.html.erb]: I think it wouldbe better to save this file.

File: app/view_objects/quizzes_view_object.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+class QuizzesViewObject
```
Comment (vovka) [inline app/view_objects/quizzes_view_object.rb]: Cover this class with tests, please.

File: app/controllers/quizzes_controller.rb [hunk 7]
```diff
@@ -4,7 +4,9 @@ class QuizzesController < ApplicationController
 
   # GET /quizzes
   def index
-    @quizzes = Quiz.all.paginate(page: params[:page], per_page: 10)
+    collection = Quiz.all.paginate(page: params[:page], per_page: 10)
```
Comment (vovka) [inline app/controllers/quizzes_controller.rb]: Cover this logic with tests, please.

File: app/view_objects/quizzes_view_object.rb (hunk 6 above)
Comment (superNazo) [inline app/view_objects/quizzes_view_object.rb]: ok

File: Gemfile (hunk 2 above)
Comment (superNazo) [inline Gemfile]: fine

File: app/controllers/quizzes_controller.rb (hunk 7 above)
Comment (superNazo) [inline app/controllers/quizzes_controller.rb]: ok

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 8]
```diff
@@ -28,9 +28,19 @@ quizzesControllers.controller('newQuizCtrl', ['$scope', '$http', '$location', 'Q
...
+  function($scope, $location, Quizzes, Quiz) {
+    $scope.quizzes;
+    $scope.total_items;
+    $scope.currentPage = 1;
 
-    $scope.quizzes = Quizzes.query();
+    $scope.setPage = function() {
+      Quizzes.query({page: $scope.currentPage}, function(data) {
+        $scope.quizzes = data.collection;
+        $scope.total_items = data.total_quizzes;
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Identifier 'total_items' is not in camel case.](https://www.codacy.com/app/vovka/video_quiz/file/2572982099/issues/source?bid=3205144&fileBranchId=3292527#l39)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 9]
```diff
@@ -28,9 +28,19 @@ quizzesControllers.controller('newQuizCtrl', ['$scope', '$http', '$location', 'Q
 
 quizzesControllers.controller('indexQuizCtrl',
   ['$scope', '$location', 'Quizzes', 'Quiz',
-  function($scope, $location, Quizzes, Quiz ) {
+  function($scope, $location, Quizzes, Quiz) {
+    $scope.quizzes;
+    $scope.total_items;
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Identifier 'total_items' is not in camel case.](https://www.codacy.com/app/vovka/video_quiz/file/2572982099/issues/source?bid=3205144&fileBranchId=3292527#l33)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js (hunk 8 above)
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Identifier 'total_items' is not in camel case.](https://www.codacy.com/app/vovka/video_quiz/file/2572982099/issues/source?bid=3205144&fileBranchId=3292527#l39)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 10]
```diff
@@ -28,9 +28,19 @@ quizzesControllers.controller('newQuizCtrl', ['$scope', '$http', '$location', 'Q
...
+  function($scope, $location, Quizzes, Quiz) {
+    $scope.quizzes;
+    $scope.totalItems;
+    $scope.currentPage = 1;
 
-    $scope.quizzes = Quizzes.query();
+    $scope.setPage = function() {
+      Quizzes.query({page: $scope.currentPage}, function(data) {
+        $scope.quizzes = data.collection;
+        $scope.totalItems = data.total_quizzes;
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Identifier 'total_quizzes' is not in camel case.](https://www.codacy.com/app/vovka/video_quiz/file/2575678245/issues/source?bid=3205144&fileBranchId=3292527#l39)

Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Identifier 'total_quizzes' is not in camel case.](https://www.codacy.com/app/vovka/video_quiz/file/2575829815/issues/source?bid=3205144&fileBranchId=3292527#l39)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 11]
```diff
@@ -28,9 +28,17 @@ quizzesControllers.controller('newQuizCtrl', ['$scope', '$http', '$location', 'Q
...
   ['$scope', '$location', 'Quizzes', 'Quiz',
-  function($scope, $location, Quizzes, Quiz ) {
+  function($scope, $location, Quizzes, Quiz) {
+    $scope.currentPage = 1;
 
-    $scope.quizzes = Quizzes.query();
+    $scope.setPage = function() {
+      Quizzes.query({page: $scope.currentPage}, function(data) {
+        $scope.quizzes = data.collection;
+        $scope.totalItems = data.total_quizzes;
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Identifier 'total_quizzes' is not in camel case.](https://www.codacy.com/app/vovka/video_quiz/file/2575845015/issues/source?bid=3205144&fileBranchId=3292527#l37)

Comment (superNazo) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: this property is generated by jbuilder which must have ruby syntax, so it should be in snake case

Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Identifier 'total_quizzes' is not in camel case.](https://www.codacy.com/app/vovka/video_quiz/file/2588273999/issues/source?bid=3205144&fileBranchId=3292527#l37)

File: app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js [hunk 12]
```diff
@@ -28,9 +28,17 @@ quizzesControllers.controller('newQuizCtrl', ['$scope', '$http', '$location', 'Q
...
   ['$scope', '$location', 'Quizzes', 'Quiz',
-  function($scope, $location, Quizzes, Quiz ) {
+  function($scope, $location, Quizzes, Quiz) {
+    $scope.currentPage = 1;
 
-    $scope.quizzes = Quizzes.query();
+    $scope.setPage = function() {
+      Quizzes.query({page: $scope.currentPage}, function(data) {
+        $scope.quizzes = data.collection;
+        $scope.totalQuizzes = data.total_items;
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Identifier 'total_items' is not in camel case.](https://www.codacy.com/app/vovka/video_quiz/file/2588308725/issues/source?bid=3205144&fileBranchId=3292527#l37)

Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: Codacy Issue found: [Identifier 'total_items' is not in camel case.](https://www.codacy.com/app/vovka/video_quiz/file/2588325231/issues/source?bid=3205144&fileBranchId=3292527#l37)

Comment (superNazo) [inline app/assets/javascripts/angularjs_app/components/quizzes/quizzesControllers.js]: this property is generated by jbuilder which must have ruby syntax, so it should be in snake case

**Pull request merged**